### PR TITLE
Removed redundant ivars in BarLineChartViewBase

### PR DESCRIPTION
**Pull request by jjatie**
In favour of proper access control

### CODE REVIEW

File: Source/Charts/Charts/BarLineChartViewBase.swift [hunk 1]
```diff
@@ -57,19 +57,28 @@ open class BarLineChartViewBase: ChartViewBase, BarLineScatterCandleBubbleChartD
     /// **default**: false
     @objc open var keepPositionOnRotation: Bool = false
     
-    /// the object representing the left y-axis
-    internal var _leftAxis: YAxis!
+    /// The left y-axis object. In the horizontal bar-chart, this is the
+    /// top axis.
+    @objc open internal(set) var leftAxis: YAxis!
```
Comment (liuxuan30) [inline Source/Charts/Charts/BarLineChartViewBase.swift]: what's the `set` purpose here?

**Comment by liuxuan30** [comment]
I'm not sure if we want to remove `_` here. I'm kind of supporting using `_` as it implies "interval", is this against latest swift trend about naming?

File: Source/Charts/Charts/BarLineChartViewBase.swift (hunk 1 above)
Comment (jjatie) [reply Source/Charts/Charts/BarLineChartViewBase.swift]: Previously there were two variables
```
internal var _leftAxis: YAxis!
@objc open var leftAxis: YAxis! {
    return _leftAxis
}
```

There is no need for this as swift provides better access control than objective-c did previously. `open internal(set)` (can also be `open private(set)` but was holding off on that one until I evaluated whether that was appropriate) makes the variable `open`, but only allows it to be set by internally.

**Comment by jjatie** [comment]
It's certainly a lot less common in Swift. In this case we don't need to have a special internal variable, as the backing ivar, and the internal ivar don't have any side effects.

File: Source/Charts/Charts/BarLineChartViewBase.swift (hunk 1 above)
Comment (liuxuan30) [reply Source/Charts/Charts/BarLineChartViewBase.swift]: wow, I don't realize swift add this already.

File: Source/Charts/Charts/BarLineChartViewBase.swift [hunk 2]
```diff
@@ -57,19 +57,28 @@ open class BarLineChartViewBase: ChartViewBase, BarLineScatterCandleBubbleChartD
...
+    /// **default**: An instance of YAxisRenderer
+    @objc open var leftYAxisRenderer: YAxisRenderer!
 
-    internal var _leftYAxisRenderer: YAxisRenderer!
-    internal var _rightYAxisRenderer: YAxisRenderer!
+    /// The right Y axis renderer. This is a read-write property so you can set your own custom renderer here.
+    /// **default**: An instance of YAxisRenderer
+    @objc open var rightYAxisRenderer: YAxisRenderer!
     
     internal var _leftAxisTransformer: Transformer!
```
Comment (liuxuan30) [inline Source/Charts/Charts/BarLineChartViewBase.swift]: how about these two and some others with `_`?  Like `_tapGestureRecognizer` It means we still need `_` for some?

Comment (jjatie) [reply Source/Charts/Charts/BarLineChartViewBase.swift]: It means I didn't touch them. I think they shouldn't have the `_`.

**Comment by liuxuan30** [comment]
Generally I'm ok with this. 
 ideas and opinions @danielgindi @petester42? @jjatie let's wait sometime to see if they have any. If you need this merged in a short time or other PRs depend on this, let me know, if like 1-2 weeks passed, we can move forward.
BTW, is this better we put this into 4.0 branch or just a minor refactor?

**Comment by jjatie** [comment]
Sounds good. This doesn't change the exposed interface for consumers, so I think it is okay to do the PR before 4.0. This and a few other PRs are setting me up for some more major changes which I will submit to 4.0 in the new year.

**Comment by jjatie** [comment]
@liuxuan30 It's been two weeks. Is it safe to merge now?

**Comment by liuxuan30** [comment]
I'm ok with this PR. But I think we should first merge the unit test PR, to make sure the demo in the future? If you manually checked the tests passed and demo builds, it's ok to 'squash and merge'

File: Source/Charts/Charts/BarLineChartViewBase.swift (hunk 1 above)
Comment (liuxuan30) [reply Source/Charts/Charts/BarLineChartViewBase.swift]: Just a moment. Have you checked if people can set axis outside from the framework? Just like other PRs similar issue?

Comment (liuxuan30) [reply Source/Charts/Charts/BarLineChartViewBase.swift]: I just did a quick check, axis is not settable outside, seems safe. Just run the tests and demo:)

**Pull request merged**